validation-test/stdlib/FixedPointConversion.swift.gyb

```diff
@@ -88,15 +112,119 @@ FixedPointConversionTraps.test("${Other}To${Self}Conversion/dest=${testValue}")
 FixedPointConversionFailure.test("${Other}To${Self}Conversion/dest=${testValue}") {
   // Test that we check if we return nil when an integer would be truncated in conversion.
   let input = get${Other}(${testValue})
-  var result = ${Self}(exactly: input)
-  expectEqual(nil, result)
-  _blackHole(result)
+  expectEmpty(${Self}(exactly: input))
 }
 %       end
 
 %     end # for testValue in ...
 %   end # for in all_integer_types (Other)
+
+%   for Other, otherSignificandBits in floatNameToSignificandBits.iteritems():
+%    otherMin = intMin(otherSignificandBits, False)
+%    otherMax = intMax(otherSignificandBits, False)
+
+%    if Other == 'Float80':
+#if arch(i386) || arch(x86_64)
+%    end
+
+%    for testValue in [repr(value) for value in [selfMin, selfMax, selfMin - 0.1, selfMax + 0.1, otherMin, otherMax, 0.0, -0.0, 0.1, -0.1]]:
+        
+%     if testValue < otherMin or testValue > otherMax: 
+%       # Can't construct `other` value to test from, do nothing and continue.
+
+%     elif testValue >= selfMin and testValue <= selfMax and testValue % 1 == 0:
+
+FloatingPointConversionTruncations.test("${Other}To${Self}Conversion/dest=${testValue}") {
+  let input = get${Other}(${testValue})
+  let result = ${Self}(input)
+  var resultConvertedBack = ${Other}(result)
+  expectEqual(${testValue}, resultConvertedBack)
+}
+
+FloatingPointConversionFailures.test("${Other}To${Self}FailableConversion/dest=${testValue}") {
+  let input = get${Other}(${testValue})
+  expectEmpty(${Self}(exactly: input))
+}
+
+%     else:
+
+%      if testValue > selfMax:
+FloatingPointConversionTraps.test("${Other}To${Self}Conversion/dest=${testValue}").crashOutputMatches(getTooLargeMessage()).code {
+expectCrashLater()
+%      elif testValue < selfMin:
+FloatingPointConversionTraps.test("${Other}To${Self}Conversion/dest=${testValue}").crashOutputMatches(getTooSmallMessage()).code {
+expectCrashLater()
+%      else:
+FloatingPointConversionTruncations.test("${Other}To${Self}Conversion/dest=${testValue}") {
+%      end
+  let input = get${Other}(${testValue})
+  var result = ${Self}(input)
+  var resultConvertedBack = ${Other}(result)
+  expectNotEqual(input, resultConvertedBack)
+}
+
+FloatingPointConversionFailures.test("${Other}To${Self}Conversion/dest=${testValue}") {
+  let input = get${Other}(${testValue})
+  expectEmpty(${Self}(exactly: input))
+}
+%    end # for in testValues
+
+// Test Always-Trapping conversions.
+
+%   if not selfSigned:
+
+FloatingPointConversionTraps.test("${Self}/${Other}/negative")
+  .crashOutputMatches(getTooSmallMessage()).code {
+  expectCrashLater()
+  _blackHole(${Self}(get${Other}(-123.0)))
+}
+
+FloatingPointConversionFailures.test("${Self}/${Other}/negative")
+expectEmpty(${Self}(exactly: get${Other}(-123.0)))
```

Comment:
Thanks for this comment! it made it really easy to track down the brace I missed on the line above.